packages/blade/src/components/Tour/_decisions/decisions.md

```diff
@@ -0,0 +1,718 @@
+  return (
+    <Box>
+      <TourStep id="step-1">
+        <Button>Click me</Button>
+      </TourStep>
+      <Box>
+        <TourStep id="step-2">
+          <Box padding="spacing.5">
+            <Text>Some content</Text>
+          </Box>
+        </TourStep>
+      </Box>
+
+      <Card>
+        <CardBody>
+          <TourStep id="step-3">
+            <Box id="step-3">
+              <Text>Some content</Text>
+              <Text>Some content</Text>
+              <Text>Some content</Text>
+            </Box>
+          </TourStep>
+        </CardBody>
+      </Card>
+    </Box>
+  );
+};
+```
+
+This component is needed because in react-native there is no `id` prop or `getElementById` platform API like web which can be used to simply query the element and highlight it.
+
+Now the open question is:
+
+1. Should we have this API difference between web & native?
+
+**A few points to consider:**
+
+- This is absolutely needed on `react-native`
+- Technically we can also expose this component for `web` too and make the API surface same, although it will largely be useless because on web we can simply use the `id` prop to query the element and highlight it.
+
+The `web` implementation will look like this:
+
+```jsx
+const TourStep = ({ id, children }) => {
+  return React.cloneElement(child, { id });
+};
+```
+
+There's also a downside to this approach on web:
+
+- Imagine if consumer wraps the `TourStep` with an `id` to a component which doesn't even accept the `id` prop, consumer won't get any type errors and `id` won't get passed to the DOM.
+
+```jsx
+// No type error :( this is a footgun for consumers
+<TourStep id="step-1">
+  <ThisComponentDoesntAccceptIdProp />
+</TourStep>
+
+// Proper type error :)
+<ThisComponentDoesntAccceptIdProp id="step-1" />
+// ^ Property 'id' does not exist on type
+```
+
+**Conclusion:**
+
+After discussing about the API differences between RN & Web,
+
+Kamlesh suggested few ways to directly use the `ref` and letting consumer attach the ref to the components, but that had few issues concerning complexity & overhead for consumers.
+
+And on the web, there was a problem that even though simply adding `id` could work, not all our components had `id` prop. That means consumers will have to wrap everything with a `Box`.
+
+- We decided **not** to go with the `id` approach on web, and instead go with the same implementation as the [react-native POC](https://github.com/razorpay/blade/compare/master...anu/tour-rn-poc#diff-4fe985a90d9ce955346ffc61e152a98272f4d02e044111d30241eb63c8fcf1b1R168-R183), where we keep track of refs of the elements via the `TourStep` component.
+  - Pros:
+    - API is now same on both RN & Web
+    - No need to wrap everything with Box, consumers can use the TourStep enhancer component.
+    - Simplified implementation, since now we don't need to maintain two types of implementation 1 for native (with refs) 1 for web (with ids).
+
+**2. Tour needs to wrap the whole app:**
```

Comment:
Its needs to wrap on screen levels right? Not on App level

Reply:
Same as https://github.com/razorpay/blade/pull/1793#discussion_r1383101447

Just kept it for future reference, check the "conclusions" below.